x-pack/plugins/observability_solution/observability/public/utils/slo/get_discover_link.ts

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { DiscoverStart } from '@kbn/discover-plugin/public';
+import { SLOWithSummaryResponse } from '@kbn/slo-schema';
+import { Filter, FilterStateStore, TimeRange } from '@kbn/es-query';
+import { isObject } from 'lodash';
+import { i18n } from '@kbn/i18n';
+import { buildEsQuery } from '../build_es_query';
+
+function createDiscoverLocator(
+  slo: SLOWithSummaryResponse,
+  showBad = false,
+  showGood = false,
+  timeRange?: TimeRange
+) {
+  const filters: Filter[] = [];
+
+  if (isObject(slo.indicator.params.filter)) {
+    slo.indicator.params.filter.filters.forEach((i) => filters.push(i));
+  }
+
+  const filterKuery = isObject(slo.indicator.params.filter)
```

Comment:
I think you can use `kqlQuerySchema.is()` or `kqlWithFiltersSchema.is()` from slo-schema.